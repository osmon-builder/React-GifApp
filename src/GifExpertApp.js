import React, {useState} from 'react'
import { AddCategory } from './Componentes/AddCategory';
import { GifGrid } from './Componentes/GifGrid';

export const GifExpertApp = ({defaultCategories = []}) => {

    // const categories = ['One Punch', 'Samuray X', 'Dragon Ball'];

    const [ categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () =>{
    //     // setCategories ([...categories, 'Naruto']) 
    //     setCategories( cats => [...cats, 'Naruto'])

    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories}/>
            <hr />

       

            <ol>
                {
                    categories.map( category =>
                         <GifGrid 
                         key={category}
                         category={category}/>
                    )
                }
            </ol>

        </>
    )
}
