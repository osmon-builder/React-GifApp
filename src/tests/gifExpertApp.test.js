import React from 'react';
import { shallow } from 'enzyme';
import {GifExpertApp}  from '../GifExpertApp';

describe('GifExpertApp', () => {

    test('match con el snapshat', () => {
        const wrapper = shallow(<GifExpertApp/>)

        expect(wrapper).toMatchSnapshot();
        
    })
    
    test('Debe de mostrar una lista de categorias', () => {

        const categories = ['One punch', ' Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>)

        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    })
    
})
