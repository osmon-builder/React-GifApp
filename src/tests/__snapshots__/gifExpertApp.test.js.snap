// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp match con el snapshat 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
